### PR TITLE
sluongng/fix macos tests

### PR DESCRIPTION
**Pull request by sluongng**
- Fix running most tests on darwin os.
- Enable `test //...` on CI with smaller caveats

---

**Version bump**: None

### CODE REVIEW

File: enterprise/server/test/integration/ci_runner/ci_runner_test.go [hunk 1]
```diff
@@ -230,7 +230,7 @@ func TestCIRunner_Push_WorkspaceWithCustomConfig_RunsAndUploadsResultsToBES(t *t
 	app := buildbuddy.Run(t)
 	runnerFlags = append(runnerFlags, app.BESBazelFlags()...)
 
-	result := invokeRunner(t, runnerFlags, []string{}, wsPath)
+	result := invokeRunner(t, runnerFlags, []string{"USER=root", "HOME=/root"}, wsPath)
```
Comment (bduffany) [inline enterprise/server/test/integration/ci_runner/ci_runner_test.go:233]: I think these should not be needed due to ensureHomeDir() and ensureUser() in `ci_runner/main.go` - what happens if these are omitted?

File: buildbuddy.yaml [hunk 2]
```diff
@@ -20,9 +20,12 @@ actions:
...
     bazel_commands:
-      # TODO: Fix `test //...` on Mac and do that instead
-      - build //enterprise/server/cmd/executor //enterprise/server //server //cli/cmd/bb --config=mac-workflows
-      - test //enterprise/server/remote_execution/... //cli/... --config=mac-workflows --test_tag_filters=-performance,-webdriver,-docker
+      - |
+        test --config=mac-workflows
+             --
+             //...
+             -//deployment:all
+             -//tools/smarter_device_manager:push_image
```
Comment (bduffany) [inline buildbuddy.yaml:28]: does it work if we mark this `push_image` target with `tags = ["manual"]`?

File: enterprise/server/test/integration/remote_execution/bazel_rbe/BUILD [hunk 3]
```diff
@@ -5,6 +5,7 @@ go_test(
     size = "small",
     srcs = ["bazel_rbe_test.go"],
     shard_count = 15,
+    target_compatible_with = ["@platforms//os:linux"],
```
Comment (bduffany) [inline enterprise/server/test/integration/remote_execution/bazel_rbe/BUILD:8]: Ditto - we probably don't want to disable this

File: enterprise/server/test/integration/ci_runner/ci_runner_test.go (hunk 1 above)
Comment (sluongng) [reply enterprise/server/test/integration/ci_runner/ci_runner_test.go:233]: I tried really hard to reproduce this from an old commit I found in my reflog... but could not.

Will revert!

File: enterprise/server/test/integration/remote_execution/BUILD [hunk 4]
```diff
@@ -9,6 +9,7 @@ go_test(
         "//enterprise/server/test/integration/remote_execution/command:testcommand",
     ],
     shard_count = 32,
+    target_compatible_with = ["@platforms//os:linux"],
```
[the remark this answers is not in the document]
Comment (sluongng) [reply enterprise/server/test/integration/remote_execution/BUILD:12]: When I tried to run these tests locally, I found that the actions were requesting `linux + amd64` executor. Tried digging around but could not find what is causing that.

> No registered executors in pool "" with os "linux" with arch "amd64"

Let me check if I could narrow down the error

File: enterprise/server/test/integration/remote_execution/bazel_rbe/BUILD (hunk 3 above)
Comment (sluongng) [reply enterprise/server/test/integration/remote_execution/bazel_rbe/BUILD:8]: ack

File: enterprise/server/util/vfs_server/vfs_server_test.go [hunk 5]
```diff
@@ -67,6 +67,10 @@ func newServer(t *testing.T) (*vfs_server.Server, string) {
 }
 
 func TestGetLayout(t *testing.T) {
+	if runtime.GOOS != "linux" {
+		// TODO(sluongng): the directory size and default symlink permission is different on MacOS
+		t.Skipf("Test assumed running on linux")
+	}
```
[the remark this answers is not in the document]
Comment (sluongng) [reply enterprise/server/util/vfs_server/vfs_server_test.go:73]: ack

File: buildbuddy.yaml (hunk 2 above)
Comment (sluongng) [reply buildbuddy.yaml]: yup, that did the magic

File: enterprise/server/test/integration/remote_execution/BUILD (hunk 4 above)
[the remark this answers is not in the document]
Comment (sluongng) [reply enterprise/server/test/integration/remote_execution/BUILD]: Managed to narrow down the error to some certain set of tests asking for linux executors.

**Review by tylerwilliams — APPROVED** [review]
LGTM

**Pull request merged**